custom/enikshay/management/commands/drtb_2_to_2_dot_1_migration.py

```diff
@@ -0,0 +1,262 @@
+            headers += self.test_case_relevant_props
+        headers.append('updates')
+        return headers
+
+    def get_case_status(self, case, episode_cases=None, recently_opened_episode_case=None, occurrence_case=None):
+        if case.type == "person":
+            return self.get_person_case_status(case)
+        elif case.type == "episode":
+            return self.get_episode_case_status(case, episode_cases)
+        elif case.type == "test":
+            return recently_opened_episode_case, self.get_test_case_status(
+                case, recently_opened_episode_case, occurrence_case)
+
+    def get_person_case_status(self, person_case):
+        person_case_props = person_case.dynamic_case_properties()
+        props_status = {'case_type': 'person', 'case_id': person_case.get_id}
+        for prop in self.person_case_relevant_props:
+            props_status[prop] = person_case_props.get(prop)
+        props_status['owner_id'] = person_case.owner_id
+        props_status['updates'] = self.person_case_updates(person_case)
+        return props_status
+
+    def get_episode_case_status(self, episode_case, episode_cases):
+        episode_case_props = episode_case.dynamic_case_properties()
+        props_status = {'case_type': 'episode', 'case_id': episode_case.get_id}
+        for prop in self.episode_case_relevant_props:
+            props_status[prop] = episode_case_props.get(prop)
+        props_status['updates'] = self.episode_case_updates(episode_case, episode_cases)
+        return props_status
+
+    def get_test_case_status(self, test_case, recently_opened_episode_case, occurrence_case):
+        test_case_props = test_case.dynamic_case_properties()
+        props_status = {'case_type': 'test', 'case_id': test_case.get_id}
+        for prop in self.test_case_relevant_props:
+            props_status[prop] = test_case_props.get(prop)
+        recently_opened_episode_case, props_status['updates'] = self.test_case_updates(
+            test_case, recently_opened_episode_case, occurrence_case)
+        return props_status
+
+    @staticmethod
+    def person_case_updates(person_case):
+        # person.case_minor_version = 21
+        props_to_update = {
+            'case_minor_version': '21',
+        }
+
+        person_case_props = person_case.dynamic_case_properties()
+
+        # If person.aadhaar_number != '', save person.aadhaar_number_obfuscated which is
+        # person.aadhaar number masked with *s except the last 4 digits
+        aadhar_num = person_case_props.get('aadhaar_number')
+        if aadhar_num:
+            props_to_update['aadhaar_number_obfuscated'] = '*' * 8 + aadhar_num[-4:]
+
+        # if person.age is blank/null and person.age_entered != blank/null
+        # then set person.age = person.age_entered.
+        # else if person.age = '' and person.dob != '', person.age = int((today() - person.dob) / 365.25)
+        if not person_case_props.get('age') and not person_case_props.get('age_entered'):
```

Comment:
@mkangia 

Should this logic be:
if not person_case_props.get('age') and ~~not~~ person_case_props.get('age_entered'):

Reply:
Right. Updating.